src/routing_node.rs

```diff
@@ -864,55 +863,45 @@ impl RoutingNode {
-        // FIXME (ben) We're sending all accepting connections as local since we don't differentiate
-        // between local and external yet.
-        // FIXME (ben 13/08/2015) We are forced to make this split as the routing message
-        // needs to contain a relay name if we are not yet connected to routing nodes
-        // under our own name.
-        if !self.core.is_connected_node() {
-            match self.get_a_bootstrap_name() {
-                Some(bootstrap_name) => {
-                    // TODO (ben 13/08/2015) for now just take the first bootstrap peer as our relay
-                    let routing_message = RoutingMessage {
-                        from_authority: Authority::Client(bootstrap_name,
-                                                          self.core.id().signing_public_key()),
-                        to_authority: Authority::ManagedNode(peer_name.clone()),
-                        content: Content::InternalRequest(InternalRequest::Connect(ConnectRequest {
-                                local_endpoints    : self.accepting_on.clone(),
-                                external_endpoints : vec![],
-                                requester_fob      : PublicId::new(self.core.id()),
-                            }
-                        )),
-                    };
-                    match SignedMessage::new(Address::Client(self.core.id().signing_public_key()),
-                                             routing_message,
-                                             self.core.id().signing_private_key()) {
-                        Ok(signed_message) => ignore(self.send(signed_message)),
-                        Err(e) => return Err(RoutingError::Cbor(e)),
-                    };
-                    Ok(())
-                }
-                None => return Err(RoutingError::NotBootstrapped),
+        let (from_authority, address) = match self.core.state() {
+            &::routing_core::State::Disconnected => return Err(RoutingError::NotBootstrapped),
+            &::routing_core::State::Bootstrapped | &::routing_core::State::Relocated => {
+                let name = match self.get_a_bootstrap_name() {
+                    Some(name) => name,
+                    // (TODO Brian 19.10.15) Shouldn't happen since we should have at least one
+                    // bootstrap connection, but should be acted on explicitly if it we get here.
+                    None => return Err(RoutingError::Interface(InterfaceError::NotConnected))
+                };
+                let signing_key = self.core.id().signing_public_key();
+                (Authority::Client(name, signing_key), Address::Client(signing_key))
+            },
+            &::routing_core::State::Terminated => {
+                // (TODO Brian 19.10.15) A new error code may be more appropriate here.
+                return Err(RoutingError::Interface(InterfaceError::NotConnected))
```

Comment:
Either way the error is wrong. Should we not create a new error for the terminated state?